app/scripts/views/sign_in.js

```diff
@@ -48,18 +48,28 @@ define(function (require, exports, module) {
     },
 
     beforeRender () {
-      this._account = this._suggestedAccount();
-
-      this._account.on('change:accessToken', () => {
-        // if no access token and password is not visible we need to show the password field.
-        if (! this._account.has('accessToken') && this.$('.password').is(':hidden')) {
-          // accessToken could be changed async by an external request after render
-          // If the ProfileClient fails to get an OAuth token with the current token then reset the view
-          this.chooserAskForPassword = true;
-          return this.render().then(function () {
-            this.setDefaultPlaceholderAvatar();
-          });
+      return p().then(() => {
```

Comment:
See if this can be merged into `_suggestedAccount` or something similar somehow.